pytests/conftest.py

```diff
@@ -47,11 +69,21 @@ def check_package(self, package):
                 return True
         return False
 
-    def remove_package(self, package):
-        ret = self.run([ 'tdnf', 'erase', '-y', package ])
-        if ret['retval'] != 0:
-            return False
-        return True
+    def erase_package(self, pkgname, pkgversion=None):
+        if pkgversion:
+            pkg = pkgname + '-' + pkgversion
+        else:
+            pkg = pkgname
+        self.run([ 'tdnf', 'erase', '-y', pkg ])
+        assert(self.check_package(pkgname) == False)
+
+    def install_package(utils, pkgname, pkgversion=None):
+        if pkgversion:
+            pkg = pkgname + '-' + pkgversion
```

Comment:
are we passing pkgversion, release, ph1/ph2/ph3 as well in the pkgversion string? or just package version? If only pkg version then it will install pkg with the same pkg version with latest release.

Reply:
No release is not being passed. Correct me if I am wrong, I was under the impression that the latest version from that release (in case of photon-3, ph3) would be picked up if no version and release is mentioned. If that's the case, then this shouldn't cause any issues and work across releases without modifications.

Reply:
If you are trying to install a specific version, tdnf expects the version number which includes the photon version (ph1/ph2/ph3) or else you will get an error saying no matching package.
Ex: tdnf install tdnf-2.0.0-10.ph3 
If you remove .ph3 from the version string, tdnf reports an error.

Reply:
We pass only `package_name` and `package_version`. that would  try to install `package_name-package_version`. Is there a problem with this?

Reply:
@keerthanakalyan, Ohh I see, then in that case the version that we pass must have the `.ph1/.ph2/.ph3` in it.

Reply:
if you are in photon3 and all your repo is pointing to ph3, then if you just mention pkg-version, then it will do it, but will install the latest release for that version.

But it will be better to include ph1/ph2/ph3 with this.

root@tkundu-photon3 [ /WS/pywbem/pywbem-0.15.0 ]# tdnf install cloud-init-18.3

Downgrading:
cloud-init               noarch       18.3-6.ph3       photon-updates   3.56M 3732716

Total installed size:   3.56M 3732716
Is this ok [y/N]:y

Downloading:
cloud-init                             1084052    100%
Testing transaction
Running transaction
Installing/Updating: cloud-init-18.3-6.ph3.noarch
Removing: cloud-init-19.1-3.ph3.noarch

Complete!
root@tkundu-photon3 [ /WS/pywbem/pywbem-0.15.0 ]#

Reply:
@tapakund: I get what you are saying. I think if we do need to mention the release, it can be part of the version string itself. `tdnf lists` command lists it that way.